### PR TITLE
[Feature][ISSUE-2401] add list api and delete api in FlinkSqlController

### LINKED ISSUE #2401: [Feature] show sql history in application detail tab
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

StreamPark saves the history of sql and config for one application, in configuration tab, we can see the history of all the configuration details, and we can also delete a specific config or compare two configs. 

image

But for sql, the detail tab only show the effective sql content.

image

 I propose to support these features that configuration tab owned for sql through two steps :
1. provide list api (get by page) and delete api in [FlinkSqlController](https://github.com/apache/incubator-streampark/blob/dev/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/FlinkSqlController.java) 
2. modify the front refer to the design of the configuration tab.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoulii**
## What changes were proposed in this pull request

Issue Number: close #2401 

## Brief change log

- add list api and delete api in FlinkSqlController

## Verifying this change

- *Manually verified the change by testing locally.* 

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/FlinkSqlController.java [hunk 1]
```diff
@@ -79,6 +81,18 @@ public RestResponse verify(String sql, Long versionId, Long teamId) {
     }
   }
 
+  @PostMapping("list")
+  public RestResponse list(Long appId, RestRequest request) {
+    IPage<FlinkSql> page = flinkSqlService.page(appId, request);
+    return RestResponse.success(page);
+  }
+
+  @PostMapping("delete")
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/FlinkSqlController.java:92]: @zhoulii Thanks for the contribution.

Does this operation need permission?

File: streampark-console/streampark-console-service/src/main/resources/mapper/core/FlinkSqlMapper.xml [hunk 2]
```diff
@@ -48,4 +48,9 @@
         where s.app_id = a.id
         and a.team_id = #{teamId}
     </select>
+
+    <select id="pageByAppId" resultType="org.apache.streampark.console.core.entity.FlinkSql" parameterType="java.lang.Long">
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/resources/mapper/core/FlinkSqlMapper.xml]: Could the lines inline into the `lambdaquery` ?

Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/resources/mapper/core/FlinkSqlMapper.xml]: nice catch, change this to lambdaquery .

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/FlinkSqlController.java (hunk 1 above)
Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/controller/FlinkSqlController.java:92]: you are right, I will add permission for this.

**Comment by zhoulii** [comment]
@RocMarshal Thanks for the review, comments are addressed, please review the changes.

File: streampark-console/streampark-console-service/src/main/assembly/script/data/mysql-data.sql [hunk 3]
```diff
@@ -122,7 +122,7 @@ insert into `t_menu` values (100071, 100033, 'link view', null, null, 'externalL
 insert into `t_menu` values (100072, 100033, 'link create', null, null, 'externalLink:create', null, '1', 1, NULL, now(), now());
 insert into `t_menu` values (100073, 100033, 'link update', null, null, 'externalLink:update', null, '1', 1, NULL, now(), now());
 insert into `t_menu` values (100074, 100033, 'link delete', null, null, 'externalLink:delete', null, '1', 1, NULL, now(), now());
-
+insert into `t_menu` values (100075, 100015, 'sql delete', null, null, 'sql:delete', null, '1', 1, null, now(), now());
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/assembly/script/data/mysql-data.sql:125]: Thanks for your contribution, we also need to put the upgrade sql into [mysql upgrade](streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql) and [pgsql upgrade](streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql)

Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/assembly/script/data/mysql-data.sql:125]: Thanks for the review and sorry for missing that, updated.

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**